examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -110,17 +91,25 @@ void BdxOtaSender::HandleTransferSessionOutput(TransferSession::OutputEvent & ev
             return;
         }
 
-        std::ifstream otaFile(mFilepath, std::ifstream::in);
-        VerifyOrReturn(otaFile.good(), ChipLogError(BDX, "%s: file read failed", __FUNCTION__));
-        otaFile.seekg(mNumBytesSent);
-        otaFile.read(reinterpret_cast<char *>(blockBuf->Start()), bytesToRead);
-        VerifyOrReturn(otaFile.good() || otaFile.eof(), ChipLogError(BDX, "%s: file read failed", __FUNCTION__));
+        if (mOnBlockQueryCallback != nullptr && mOnBlockQueryCallback->mCall != nullptr)
+        {
+            mOnBlockQueryCallback->mCall(mOnBlockQueryCallback->mContext, blockBuf, blockData.Length, blockData.IsEof,
+                                         mNumBytesSent);
+        }
+        else
+        {
+            mTransfer.AbortTransfer(StatusCode::kUnknown);
+            return;
+        }
+
+        if (blockData.Length == 0 && blockData.IsEof == false)
```

Comment:
What about having `mOnBlockQueryCallback` return a status instead of trying to interpret error conditions from `blockData`?

Reply:
Sound right, changed, now callback should return an error if something goes wrong.